torchtune/utils/_profiler.py

```diff
@@ -6,28 +6,41 @@
 
 import contextlib
 
-from typing import ContextManager, Optional
+import os
+import time
+from functools import partial
+from pathlib import Path
+from typing import ContextManager, Optional, Tuple
 
 import torch
-from torch.profiler import profile
+import torch.distributed
+
+from omegaconf import DictConfig
+from torch._C._profiler import _ExperimentalConfig
+from torch.profiler import profile, tensorboard_trace_handler
+from torchtune.utils import get_world_size_and_rank
+
+from torchtune.utils.logging import get_logger
+
+log = get_logger("INFO")
 
 
 def profiler(
```

Comment:
It is not clear to me the relationship between this method and setup_torch_profiler, is this now deprecated? If so, let's just remove it entirely

Reply:
Removed